docs/hole-finder.py

```
@@ -0,0 +1,77 @@
+
+    print("======================================================================================\n")
+
+    print(
+        "Locating hole... Disregard if you are concerned with any other situation than a defect with a SINGLE HOLE\n")
+    print(
+        "Information on EINT min and max can be ignored if you are using pymatgen to make your PARCHG file")
+
+    hole_spin_down = df[(df["up-occupancy"] == 1.0) & (df["down-occupancy"] == 0.0)]
+    hole_spin_up = df[(df["up-occupancy"] == 0.0) & (df["down-occupancy"] == 1.0)]
+
+    if hole_spin_up.empty == True and hole_spin_down.empty == False:
```

Ruff (0.11.9)

Avoid equality comparisons to `True`; use `if hole_spin_up.empty:` for truth checks

Replace with `hole_spin_up.empty`

(E712)